pkg/cache/internal/cache_reader.go

```diff
@@ -73,14 +73,20 @@ func (c *CacheReader) Get(_ context.Context, key client.ObjectKey, out runtime.O
 	// TODO(directxman12): revisit the decision to always deepcopy
 	obj = obj.(runtime.Object).DeepCopyObject()
 
-	// Copy the value of the item in the cache to the returned value
-	// TODO(directxman12): this is a terrible hack, pls fix (we should have deepcopyinto)
-	outVal := reflect.ValueOf(out)
-	objVal := reflect.ValueOf(obj)
-	if !objVal.Type().AssignableTo(outVal.Type()) {
-		return fmt.Errorf("cache had type %s, but %s was asked for", objVal.Type(), outVal.Type())
+	oType := reflect.TypeOf(obj)
+	_, ok := oType.MethodByName("DeepCopyInto")
+	if ok {
+		reflect.ValueOf(obj).MethodByName("DeepCopyInto").Call([]reflect.Value{reflect.ValueOf(out)})
```

Comment:
Do we need to check for an error here?

Reply:
Actually, I don't think we do, 

https://github.com/kubernetes/api/blob/master/admission/v1beta1/zz_generated.deepcopy.go#L29

Does that sound correct?